jena-base/src/main/java/org/apache/jena/atlas/iterator/Iter.java

```diff
@@ -29,6 +29,14 @@
 import org.apache.jena.atlas.lib.Closeable ;
 import org.apache.jena.atlas.lib.Sink ;
 
+/**
+ * Iter provides utilities for working with {@link Iterator}s.
+ *
+ * Iter should never be used as a return type or parameter type in the public contract of a class. It is only to be used
+ * inside implementation code and is instantiated only to allow method-chaining as part of a calculation.
```

Comment:
Yes, that's why the note is useful. If `Iter` actually imposed that in the code there would be no need for documentation.

Is there in fact any difference between `Iter` and `ExtendedIterator`? They are both subtypes of `java.util.Iterator` that provide convenient methods, with the functionality of `Iter` entirely overlapping that of `ExtendedIterator`. I've tried repeatedly to discover this information, and my experience shows that it is quite possible to confuse them. You were quite explicit in explaining to me in connection with JENA-966 that they are not to be used in the same circumstances, and all I'm trying to do here is determine what those circumstances actually are and record them.